lib/jekyll/filters/url_filters.rb

```diff
@@ -33,6 +34,20 @@ def relative_url(input)
         ).normalize.to_s
       end
 
+      # Produces a path relative to the current page. For example, if the path
+      # `/images/me.jpg` is supplied, then the result could be, `me.jpg`,
+      # `images/me.jpg` or `../../images/me.jpg` depending on whether the current
+      # page lives in `/images`, `/` or `/sub/dir/` respectively.
+      #
+      # input - a path relative to the project root
+      #
+      # Returns the supplied path relativized to the current page.
+      def relativize_url(input)
+        pageUrl = @context.registers[:page]["url"]
+        pageDir = Pathname(pageUrl).parent
+        Pathname(input).relative_path_from(pageDir).to_s
+      end
```

Comment:
if you check the existing URLFilters, you'll see that there's a guard against `nil` inputs..
`Pathname(input).relative_path_from(pageDir).to_s` will raise if `input` is `nil`

Perhaps alternatively stringify the input beforehand?
```diff
- `Pathname(input).relative_path_from(pageDir).to_s`
+ Pathname(input.to_s).relative_path_from(pageDir).to_s
```